src/main/java/data/SubjectRawData.java

```diff
@@ -0,0 +1,27 @@
+package data;
+
+import java.io.File;
+import java.util.SortedMap;
+
+/**
+ * Created by Timotej on 21-Oct-15.
+ */
+public final class SubjectRawData {
```

Comment:
The reason I made it final is so that it is a data holding object and I wanted to limit it as much as possible. I don't think it should implement the visitable because it represents just 1 Subject. A collection of these is the visitable I think, so perhaphs the Filesystem Database.